kong/tools/utils.lua

```diff
@@ -1566,4 +1567,75 @@ _M.sha256_hex       = sha256_hex
 _M.sha256_base64    = sha256_base64
 _M.sha256_base64url = sha256_base64url
 
+
+-----------------------------------------------------------------------------
+-- From https://github.com/lunarmodules/luasocket/blob/master/src/url.lua
+-- Parses a url and returns a table with all its parts according to RFC 2396
+-- The following grammar describes the names given to the URL parts
+-- <url> ::= <scheme>://<authority>/<path>;<params>?<query>#<fragment>
+-- <authority> ::= <userinfo>@<host>:<port>
+-- <userinfo> ::= <user>[:<password>]
+-- <path> :: = {<segment>/}<segment>
+-- Input
+--   url: uniform resource locator of request
+--   default: table with default values for each field
+-- Returns
+--   table with the following fields, where RFC naming conventions have
+--   been preserved:
+--     scheme, authority, userinfo, user, password, host, port,
+--     path, params, query, fragment
+-- Obs:
+--   the leading '/' in {/<path>} is considered part of <path>
+-----------------------------------------------------------------------------
+_M.parse_url = function(url, default)
```

Comment:
Can't we use https://github.com/ledgetech/lua-resty-http/blob/master/lib/resty/http.lua#L255 instead of copying?

Reply:
that one doesn't support credential inside URI, it will need change. instead of bringing uncertainty,
i would suggest go with known-to-be good solution.